pkg/reconciler/build/resources/pod_test.go

```diff
@@ -480,7 +488,15 @@ func TestMakePod(t *testing.T) {
 			}
 
 			if d := cmp.Diff(&got.Spec, c.want, ignorePrivateResourceFields); d != "" {
-				t.Errorf("Diff:\n%s", d)
+				t.Errorf("Diff spec:\n%s", d)
+			}
+
+			if c.bAnnotations != nil {
```

Comment:
nit: This is just a suggestion

```
wantAnnotations := map[string]string{"sidecar.istio.io/inject": "false"}  // this indicates that pod annotation will always have this annotation set as default
if c.bAnnotations != nil {
	for key, val := range c.bAnnotations { // adding this loop suggests new set of keys will be added to default set
		wantAnnotations[key] = val 
	}
}
if d := cmp.Diff(got.Annotations, wantAnnotations); d != "" {
	t.Errorf("Diff annotations:\n%s", d)
}
```

I prefer this change because annotations will be checked on all test cases not just the one which has custom annotations set.